incubator/node-red/image/Dockerfile-stack

```diff
@@ -1,13 +1,16 @@
 FROM node:12
+  
+RUN useradd -m node_user && groupadd node_group
```

Comment:
how do I reproduce the problem that led to this being the solution?

Reply:
For me the only way I could reproduce the problem was by switching the `node_user` to `node`. It doesn't really make much sense but I'd like to retest some of these changes again (since its almost been a month since opening). Just out of interest, which OS are you on? I only hit these problems on my Ubuntu VM.

Reply:
I am on an Ubuntu laptop.

Reply:
I cannot repro any problems with builtin user, I checked out this PR, did this:
```
incubator/node-red (pr/624 *$%) % git diff
diff --git a/incubator/node-red/image/Dockerfile-stack b/incubator/node-red/image/Dockerfile-stack
index 291ac94..76f47e1 100644
--- a/incubator/node-red/image/Dockerfile-stack
+++ b/incubator/node-red/image/Dockerfile-stack
@@ -1,6 +1,4 @@
 FROM node:12
-  
-RUN useradd -m node_user && groupadd node_group
 
 ENV APPSODY_MOUNTS=/:/project/user-app
 ENV APPSODY_DEPS=/project/user-app/node_modules
@@ -22,12 +20,12 @@ ENV APPSODY_DEBUG_KILL=true
 
 ENV APPSODY_TEST="npm test"
 
-COPY --chown=node_user:node_group ./LICENSE /licenses/
-COPY --chown=node_user:node_group ./project /project
-COPY --chown=node_user:node_group ./config /config
+COPY --chown=node:node ./LICENSE /licenses/
+COPY --chown=node:node ./project /project
+COPY --chown=node:node ./config /config
 
 WORKDIR /project
-USER node_user
+USER node
 
 RUN mkdir -p /project/user-app/node_modules
 RUN npm install && npm audit fix
@@ -38,4 +36,4 @@ ENV NPM_CONFIG_CACHE=/project/user-app/.npm
 ENV XDG_CONFIG_HOME=/project/user-app/.config
 
 EXPOSE 3000
-EXPOSE 9229
\ No newline at end of file
+EXPOSE 9229
```

packaged the stack, init and run, no issues.

```

incubator/ex (pr/624 *$%) % appsody run
Running development environment...
Using local cache for image dev.local/appsody/node-red:0.1
Running command: docker run --rm -p 3000:3000 -p 9229:9229 --name ex-dev -u 1000:1000 -e APPSODY_USER=1000 -e APPSODY_GROUP=1000 -v /home/sam/w/cloud-node/appsody-stacks/incubator/ex/:/projec
t/user-app -v ex-deps:/project/user-app/node_modules -v appsody-controller-0.3.4:/.appsody -t --entrypoint /.appsody/appsody-controller dev.local/appsody/node-red:0.1 --mode=run
[Container] Running APPSODY_PREP command: npm install --prefix user-app && npm audit fix --prefix user-app
npm WARN SampleAppsody-Node-RED-Project@0.1.1 No repository field.
[Container] npm WARN SampleAppsody-Node-RED-Project@0.1.1 No license field.
[Container] 
up to date in 0.293s
[Container] found 0 vulnerabilities
[Container] 
npm WARN SampleAppsody-Node-RED-Project@0.1.1 No repository field.
[Container] npm WARN SampleAppsody-Node-RED-Project@0.1.1 No license field.
[Container] 
[Container] up to date in 0.04s
[Container] fixed 0 of 0 vulnerabilities in 0 scanned packages
[Container] Running command:  npm start
[Container] 
[Container] > node-red-stack@0.1.1 start /project
[Container] > node server.js
[Container] 
[Container] [Mon Mar  2 23:15:45 2020] com.ibm.diagnostics.healthcenter.loader INFO: Node Application Metrics 5.1.1.202003022308 (Agent Core 4.0.5)
[Container] [Mon Mar  2 23:15:46 2020] com.ibm.diagnostics.healthcenter.mqtt INFO: Connecting to broker localhost:1883
[Container] 2 Mar 23:15:46 - [info] 
[Container] 
[Container] Welcome to Node-RED
[Container] ===================
[Container] 
[Container] 2 Mar 23:15:46 - [info] Node-RED version: v1.0.4
[Container] 2 Mar 23:15:46 - [info] Node.js  version: v12.16.1
[Container] 2 Mar 23:15:46 - [info] Linux 5.3.0-40-generic x64 LE
[Container] 2 Mar 23:15:46 - [info] Loading palette nodes
[Container] 2 Mar 23:15:47 - [info] Context store  : 'default' [module=memory]
[Container] 2 Mar 23:15:47 - [info] User directory : /project/user-app/
[Container] 2 Mar 23:15:47 - [warn] Projects disabled : set editorTheme.projects.enabled=true to enable
[Container] 2 Mar 23:15:47 - [info] Flows file     : /project/user-app/flow.json
[Container] 2 Mar 23:15:47 - [info] Server now running at 
[Container] 2 Mar 23:15:47 - [warn] 
[Container] 
[Container] ---------------------------------------------------------------------
[Container] Your flow credentials file is encrypted using a system-generated key.
[Container] 
[Container] If the system-generated key is lost for any reason, your credentials
[Container] file will not be recoverable, you will have to delete it and re-enter
[Container] your credentials.
[Container] 
[Container] You should set your own key using the 'credentialSecret' option in
[Container] your settings file. Node-RED will then re-encrypt your credentials
[Container] file using your chosen key the next time you deploy a change.
[Container] ---------------------------------------------------------------------
[Container] 
[Container] 2 Mar 23:15:47 - [info] Starting flows
[Container] 2 Mar 23:15:47 - [info] Started flows
```

permissions look correct:

```
cloud-node/appsody-stacks (pr/624 *$%) % docker exec -it d256c14ad8af ls -l /project /project/user-app
/project:
total 224
-rw-rw-r--   1 node node   1231 Mar  2 23:07 Dockerfile
drwxr-xr-x 427 node node  20480 Mar  2 23:08 node_modules
-rw-r--r--   1 node node 184832 Mar  2 23:08 package-lock.json
-rw-rw-r--   1 node node    675 Mar  2 23:07 package.json
-rw-r--r--   1 node node      0 Mar  2 23:15 resourceRegistry.log
-rw-rw-r--   1 node node   2686 Mar  2 23:07 server.js
drwxrwxr-x   2 node node   4096 Mar  2 23:07 test
drwxrwxr-x   6 node node   4096 Mar  2 23:15 user-app

/project/user-app:
total 24
-rw-rw-r-- 1 node node 1200 Mar  2 23:15 flow.json
-rw-rw-r-- 1 node node    3 Mar  2 23:15 flow_cred.json
drwxr-xr-x 3 node node 4096 Mar  2 23:15 lib
drwxr-xr-x 2 node node 4096 Mar  2 23:07 node_modules
-rw-rw-r-- 1 node node   99 Mar  2 23:15 package-lock.json
-rw-rw-r-- 1 node node  308 Mar  2 23:15 package.json
```